### DISCUSSION TITLE
Help for use of imagebutton in berry

### DISCUSSION THREAD

**Discussion by fdaubercy** [arendst/Tasmota#22123] [Q&A] [answered]
Hello.
I use tasmota with LVGL on a ILI9488 screen.
I'm trying to set an image button with berry.
I can't find the code needed to display on this screen an image button with the 'imagebutton' function with an image stored in memory.
Can someone help me?
Thank you in advance to all.

**Comment by s-hadinger** [comment]
Hi. Are you using native lvgl binding or HASPmota?
Keep in mind that images take a lot of memory and can be problematic without psram. 

There is `set_src` method that takes 3 arguments for left/mid/right images. Each arg can be either a pointer to ab image in Flash (probably not your case) or a string containing the path of the image in the file system. 

I didn't try it so I don't have test code for it

**Comment by fdaubercy** [comment, marked as answer]
Good morning.

I use an ESP32S3 with 16MB of psram.
The image is stored in flash memory (filesystem).

Here is the code I use:

```
import global
lv.start()
hres = lv.get_hor_res()       # should be 320
vres = lv.get_ver_res()       # should be 240
var scr = lv.scr_act()            # default screean object
f20 = lv.montserrat_font(20)  # load embedded Montserrat 20

var btn_style = lv.style()
btn_style.set_radius(10)                        # radius of rounded corners
#btn_style.set_bg_opa(lv.OPA_COVER)              # 100% background opacity
if f20 != nil btn_style.set_text_font(f20) end  # set font to Montserrat 20
# btn_style.set_bg_color(lv.color(0x1fa3ec))      # background color #1FA3EC (Tasmota Blue)
btn_style.set_border_color(lv.color(0x0000FF))  # border color #0000FF
btn_style.set_text_color(lv.color(0xFFFFFF))    # text color white #FFFFFF

btn = lv.imagebutton(scr)
btn.add_style(btn_style, lv.PART_MAIN | lv.STATE_DEFAULT) 
btn.set_pos(30, 30)
btn.set_size(100, 49)
btn.set_src(lv.IMAGEBUTTON_STATE_RELEASED, None, "/right_button.png", None)
btn.set_state(lv.IMAGEBUTTON_STATE_RELEASED)
```

But i see nothing on the creen.

**Reply by s-hadinger** [reply]
I can't test right now. But keep in mind that LVGL imposes that there is a drive letter.

Can you try with:
`btn.set_src(lv.IMAGEBUTTON_STATE_RELEASED, None, "A:/right_button.png", None)`

**Comment by fdaubercy** [comment]
Good morning.
Thank you, it was indeed the letter 'A:' before the slash in the image file path.
But also, the png image was too heavy. I understand why, imagebutton has 3 images as parameter.
This allowed me to reduce the weight of each part.
Thanks again.
Frederic
